tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/listener/rag/LlmMailPrioritizationClassifierListener.java

```diff
@@ -0,0 +1,258 @@
+        You are an email-triage classifier. Your ONLY task is to evaluate whether an incoming email requires immediate human action by the recipient.
+        Return strictly one word: "YES" or "NO". Do not add explanations or any additional text.
+        An email requires immediate action (YES) only if it reports urgent issues (like production failures or security incidents), time-sensitive requests or approvals (with clear deadlines), or payment/billing problems that might lead to service disruption soon.
+        Return NO for newsletters, marketing, spam, general updates, casual conversations, or unclear/ambiguous messages. If the email looks like spam or phishing, always return NO.
+        """;
+
+    private final MailboxManager mailboxManager;
+    private final MessageIdManager messageIdManager;
+    private final SystemMailboxesProvider systemMailboxesProvider;
+    private final StreamingChatLanguageModel chatLanguageModel;
+    private final HtmlTextExtractor htmlTextExtractor;
+    private final MetricFactory metricFactory;
+    private final String systemPrompt;
+    private final int maxBodyLength;
+
+    @Inject
+    public LlmMailPrioritizationClassifierListener(MailboxManager mailboxManager,
+                                                   MessageIdManager messageIdManager,
+                                                   SystemMailboxesProvider systemMailboxesProvider,
+                                                   StreamingChatLanguageModel chatLanguageModel,
+                                                   HtmlTextExtractor htmlTextExtractor,
+                                                   MetricFactory metricFactory,
+                                                   HierarchicalConfiguration<ImmutableNode> configuration) {
+        this.mailboxManager = mailboxManager;
+        this.messageIdManager = messageIdManager;
+        this.systemMailboxesProvider = systemMailboxesProvider;
+        this.chatLanguageModel = chatLanguageModel;
+        this.htmlTextExtractor = htmlTextExtractor;
+        this.metricFactory = metricFactory;
+
+        this.systemPrompt = Optional.ofNullable(configuration.getString("listener.configuration." + SYSTEM_PROMPT_PARAM, null))
+            .filter(s -> !s.isBlank())
+            .orElse(DEFAULT_SYSTEM_PROMPT);
+        this.maxBodyLength = configuration.getInt("listener.configuration." + MAX_BODY_LENGTH_PARAM, DEFAULT_MAX_BODY_LENGTH);
+        Preconditions.checkArgument(maxBodyLength > 0, "'maxBodyLength' must be strictly positive");
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public boolean isHandling(Event event) {
+        return event instanceof MailboxEvents.Added added && added.isAppended();
+    }
+
+    @Override
+    public Publisher<Void> reactiveEvent(Event event) {
+        if (event instanceof MailboxEvents.Added added && added.isAppended()) {
```

Comment:
```suggestion
        if (event instanceof MailboxEvents.Added added && added.isDelivery()) {
```

In order to be very precise...